force-app/main/service/DonationHistoryService.cls

```diff
@@ -84,9 +85,19 @@ public with sharing class DonationHistoryService {
     */
     public List<Opportunity> getDonationHistory() {
         if (hasAccessToDonationHistory()) {
-            return selector.getWonOpportunitiesByContactId(this.contactId);
+            return selector.getWonOpportunitiesByContactId(this.contactId, this.offset);
         } else {
             throw new UTIL_Permissions.InsufficientPermissionException(Label.commonInsufficientPermissions);
         }
     }
+    
+    /** 
+    * @author Salesforce.org
+    * @date 2021
+    * @description method to return Donation History Number of Records for infinite scroll.
+    * @param contactId contact id which is going to be requested.
+    */
+    public List<AggregateResult> getTotalNumberOfRecords() {
```

Comment:
I think this service should return the number(Integer) and not the AggregateResult.

Reply:
done